tests/rptest/tests/services_self_test.py

```diff
@@ -50,16 +50,21 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, num_brokers=3, **kwargs)
 
     @cluster(num_nodes=4)
-    def test_kgo_repeater(self):
-        topic = 'test'
-        self.client().create_topic(
-            TopicSpec(name=topic,
-                      partition_count=16,
-                      retention_bytes=16 * 1024 * 1024,
-                      segment_bytes=1024 * 1024))
+    @matrix(topic_spec=[
+        TopicSpec(name='test_with_lifetime_as_bytes',
```

Comment:
The tests in `services_self_test.py` are here to test the helper services, not Redpanda -- please add the tests for segment.ms elsewhere.